src/writer.js

```diff
@@ -67,7 +67,10 @@ define([
             jrm = "http://dev.commcarehq.org/jr/xforms";
         }
 
-        uuid = form.formUuid; //gets set at parse time/by UI
+        uuid = form.formUuid;
+        if(!uuid) {
+            uuid = "http://openrosa.org/formdesigner/" + util.generate_xmlns_uuid();
```

Comment:
Don't you need to write the value back to `form.formUuid`? Otherwise it will generate a new UUID each time the form is written. Might be worth making a test for that scenario.